types/eslint/index.d.ts

```diff
@@ -1189,7 +1173,7 @@ export namespace Linter {
             };
         });
 
-    type ParserModule =
+    type LegacyParser =
```

Comment:
I'm not sure there should be a difference between `Parser` and `LegacyParser`. Both flat config and eslintrc use the same parsers.

Reply:
Good point. It seems that `Linter.Parser` (previously `Linter.FlatConfigParserModule`) was introduced to allow using parsers that return a non-ESTree AST (#68232). So `Linter.Parser` is for typing generic parsers and `Linter.LegacyParser` (previously `Linter.Parser`) is for ESTree parsers. I realize the naming makes no sense. Maybe we could rename `Linter.LegacyParser` to `Linter.ESTreeParser` and keep the name `Linter.LegacyParser` for generic parsers?

Reply:
Maybe `Linter.NonESTreeParser` and `Linter.ESTreeParser`?

Reply:
> Maybe `Linter.NonESTreeParser` and `Linter.ESTreeParser`?

Makes sense. We could use `Linter.Parser` as a union of both:
```ts
type Parser = NonESTreeParser | ESTreeParser;
```